### ISSUE TITLE
Support fixtures defined in other files/plugins

### ISSUE THREAD

**Issue by kasium** [DanielNoord/pylint-pytest-plugin#16]
It would be great if this plugin would understand fixtures defined in plugins and other files.

**test_foo.py**
```py
def test(my_fixture):
    assert True
```

**conftest.py**
```py
import pytest

@pytest.fixture
def my_fixture():
    # do some mocking
    return 42
```

Currently this is not supported tp supress errors generated by `my_fixture`. Note that, conftest.py can exist also in every parent directory and therefore be nested. Also plugins might provide plugins.

Therefore asking pytest like `pytest --fixtures --collect-only test_foo` would be good (similar to pylint_pytest)

**Comment by DanielNoord** [comment]
@kasium I have (finally) taken a look at the old plugin and see that it works by actually invoking `pylint` to collect all fixtures. I'm not sure if I agree with that approach as it means that `pylint` will be even slower and reliant on the behaviour of another tool.

**Comment by kasium** [comment]
Yes, that is one downside of the plugin but I see no way around that if you want a stable and reliable plugin.
However, what do you think about a persistent cache for the fixtures? So that these at only fetched with the first run or if the file was changed. But still a special handling for these conftest files would be needed...

**Comment by DanielNoord** [comment]
I think it might be quite hard to set up the correct cache hit and cache regenerate conditions. `pylint` itself notoriously doesn't do this as well because we deemed it to be too hard.

**Comment by kasium** [comment]
Valid point. I thought also about it and I guess for normal projects, a simple cache would work. However, when it comes to dynamic plugins, imports in conftest, etc. it won't work or would be too complicated.

However, here are two ideas:
1. Having a "fast" and "slow" mode --> The "fast" mode would work with your current code and the "slow" one would ask pytest for the fixtures
2. (could be combined with 1.) --> ask the pytest team to provide a way to receive all possible fixtures in all files. It would nto be 100% safe (e.g. a unused-variable is in fact unused and not a fixtures) but it's still better than nothing. The collection needs only to be done once per pylint session and not for every file.

What do you think?

**Comment by kasium** [comment]
I guess asking pytest with the "--fixtures" flag should return all fixtures defined in the current scope. The previous plugin did it on file level.and not down to the single test function as far as I known

**Comment by DanielNoord** [comment]
> I guess asking pytest with the "--fixtures" flag should return all fixtures defined in the current scope. The previous plugin did it on file level.and not down to the single test function as far as I known

Yeah, that's what I'm worried about. Because then the next opened issue will be: why is the support for fixtures broken for my function, etc. I think we would open a can of worms with it.

**Comment by kasium** [comment]
Fair point, I'm really not sure how to continue then
